### PR TITLE
[full-ci][gui-tests] update docker images

### PR DESCRIPTION
**Pull request by saw-jan**
Updated docker images:
`owncloudci/client:fedora-38-amd64` -> `owncloudci/client:fedora-39-amd64`
`owncloudci/squish:fedora-38-7.1.1-qt64x-linux64` -> `owncloudci/squish:fedora-39-7.2.1-qt66x-linux64`

Fixes https://github.com/owncloud/client/issues/11597

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -235,6 +235,10 @@ def build_client(image = OC_CI_CLIENT, ctest = True):
     cmake_options = '-G"%s" -DCMAKE_C_COMPILER="%s" -DCMAKE_CXX_COMPILER="%s" -DCMAKE_BUILD_TYPE="%s"'
```
Comment (TheOneRing) [inline .drone.star:235]: Can you use /usr/clang /usr/clang++ (else it uses the clang provided by craft)

Comment (saw-jan) [reply .drone.star]: Now I only used `cmake -G"Ninja" -DBUILD_TESTING=OFF -S ..` while building client using craft installed libraries

**Comment by saw-jan** [comment]
Most of the tests passed. remaining failures are due to object error
https://drone.owncloud.com/owncloud/client/18154/5/11

**Comment by TheOneRing** [comment]
> Most of the tests passed. remaining failures are due to object error https://drone.owncloud.com/owncloud/client/18154/5/11

We now use qml for the spaces browser here too, if a object name or property is missing in the code please tell me.

**Comment by saw-jan** [comment]
> > Most of the tests passed. remaining failures are due to object error https://drone.owncloud.com/owncloud/client/18154/5/11
> 
> We now use qml for the spaces browser here too, if a object name or property is missing in the code please tell me.

okey, the failure was due to the change in the selector type

**Comment by TheOneRing** [comment]
Please update https://github.com/owncloud/client/blob/5b5b25557840bc273fffa38613d8f7937b484091/.drone.star#L72-L76
to use /usr/bin/clang else it uses the clang from the craft environment

File: .drone.star [hunk 2]
```diff
@@ -232,8 +232,11 @@ def gui_test_pipeline(ctx):
     return pipelines
 
 def build_client(image = OC_CI_CLIENT, ctest = True):
-    cmake_options = '-G"%s" -DCMAKE_C_COMPILER="%s" -DCMAKE_CXX_COMPILER="%s" -DCMAKE_BUILD_TYPE="%s"'
-    cmake_options = cmake_options % (build_config["generator"], build_config["c_compiler"], build_config["cxx_compiler"], build_config["build_type"])
+    cmake_options = '-G"%s"' % build_config["generator"]
+
+    if image != OC_CI_SQUISH:
```
Comment (TheOneRing) [inline .drone.star:237]: Why the change here?

Comment (saw-jan) [reply .drone.star:237]: For [ctest pipeline](https://drone.owncloud.com/owncloud/client/18160/4/2), I am using `owncloudci/client:fedora-39-amd64` which doesn't have to craft environment. so maintaining the previous build configs if not using `OC_CI_SQUISH` which has craft environment

**Comment by saw-jan** [comment]
> Please update
> 
> https://github.com/owncloud/client/blob/5b5b25557840bc273fffa38613d8f7937b484091/.drone.star#L72-L76
> 
> 
> to use /usr/bin/clang else it uses the clang from the craft environment

these configs are only used when not using `OC_CI_SQUISH` (image with craft environment). So should be fine, right?

**Comment by TheOneRing** [comment]
> > Please update
> > https://github.com/owncloud/client/blob/5b5b25557840bc273fffa38613d8f7937b484091/.drone.star#L72-L76
> > 
> > to use /usr/bin/clang else it uses the clang from the craft environment
> 
> these configs are only used when not using `OC_CI_SQUISH` (image with craft environment). So should be fine, right?

I don't really see a reason to no longer provide them, also passing the build type is always a good idea and you removed that along with the compiler.

**Comment by TheOneRing** [comment]
(Also can you clean up the commit history, or should I squash the pr?)

**Review by Salipa-Gurung — DISMISSED** [review]

**Comment by saw-jan** [comment]
> (Also can you clean up the commit history, or should I squash the pr?)

we usually do `Squish and merge`

**Comment by saw-jan** [comment]
>I don't really see a reason to no longer provide them, also passing the build type is always a good idea and you removed that along with the compiler.

okey, I will add the build_type on every build

**Review by TheOneRing — APPROVED** [review]
